src/buildSelectors.ts

```diff
@@ -62,23 +63,33 @@ export function buildSelectors<InternalQueryArgs, Definitions extends EndpointDe
 
   return { buildQuerySelector, buildMutationSelector };
 
+  function withRequestFlags(substate: CombinedState<any>): any {
+    return substate?.status
+      ? {
+          ...substate,
+          ...getRequestStatusFlags(substate.status),
+        }
+      : substate;
+  }
+
   function buildQuerySelector(
     endpoint: string,
     definition: QueryDefinition<any, any, any, any>
   ): QueryResultSelector<any, RootState> {
     return (arg?) => (rootState: RootState) =>
       (arg === skipSelector
```

Comment:
This could now just be wrapped as a whole, like 
```ts
      withRequestFlags(
        (arg === skipSelector
          ? undefined
          : (rootState[reducerPath] as InternalState).queries[serializeQueryArgs(definition.query(arg), endpoint)]) ??
          defaultQuerySubState
      );
```

BUT: I'd say we start to need reselect here as the result would become unstable otherwise. Mabe something like

```ts
  function selectInternalState(rootState: RootState) {
    return rootState[reducerPath] as InternalState;
  }

  function buildQuerySelector(
    endpoint: string,
    definition: QueryDefinition<any, any, any, any>
  ): QueryResultSelector<any, RootState> {
    return (arg?) => {
      const selectQuerySubState = createSelector(
        selectInternalState,
        (internalState) =>
          (arg === skipSelector
            ? undefined
            : internalState.queries[serializeQueryArgs(definition.query(arg), endpoint)]) ?? defaultQuerySubState
      );
      return createSelector(selectQuerySubState, withRequestFlags);
    };
  }
```

The problem is: we'll also need some to start using the selectors differently in `useQuery/useMutation`.

So in `buildQueryHook`, it was
```ts
      const querySelector = querySelectors[name];
      const currentState = useSelector(querySelector(skip ? skipSelector : stableArg));
```
and should probably become
```ts
      const buildQuerySelector = querySelectors[name];
      const querySelector = useMemo(() => buildQuerySelector(skip ? skipSelector : stableArg), [skip, stableArg]);
      const currentState = useSelector(querySelector);
```

same goes for the mutation stuff.

An alternative to that would be using `shallowEquals` in `useSelector` and keeping this as-is.
But long-term I *assume* that this is the better direction to go.
Maybe we should rename `selectors` to `getSelector` on the api return object?

Reply:
Hmm, regarding the naming, I don't really have an opinion. I think keeping it as `selectors` is fine.

Reply:
I mean, those are really more selector factories than selectors though, so I'm not really happy with just `selectors`.
@markerikson do you have an opinion there?

Reply:
Only partly following along because I'm working, so not really :)  other than "shorter is good", but also "accurate is good"